hugr-py/src/hugr/val.py

```diff
@@ -50,13 +50,16 @@ def bool_value(b: bool) -> Sum:
 
 
 @dataclass
-class Tuple(Value):
+class Tuple(Sum):
     vals: list[Value]
 
-    def type_(self) -> tys.Tuple:
-        return tys.Tuple(*(v.type_() for v in self.vals))
+    def __init__(self, *vals: Value):
+        val_list = list(vals)
+        super().__init__(
+            tag=0, typ=tys.Tuple(*(v.type_() for v in val_list)), vals=val_list
+        )
 
-    def to_serial(self) -> sops.TupleValue:
+    def to_serial(self) -> sops.TupleValue:  # type: ignore[override]
```

Comment:
Any idea why the ignore is needed?
It seems straightforward code

Reply:
`sops.TupleValue` isn't an instance of `sops.SumValue` so mypy doesn't like the override of `Sum.to_serial`

Reply:
will add comment